library/kotlin/src/io/envoyproxy/envoymobile/Client.kt

```diff
@@ -10,29 +10,29 @@ interface Client {
    * @param responseHandler the callback for receiving stream events.
    * @return the emitter for streaming data outward.
    */
-  fun startStream(request: Request, responseHandler: ResponseHandler): StreamEmitter
+  fun send(request: Request, responseHandler: ResponseHandler): StreamEmitter
 
 
   /**
    * Convenience function for sending a unary request.
    *
    * @param request  The request to send.
-   * @param body Serialized data to send as the body of the request.
+   * @param data Serialized data to send as the body of the request.
    * @param trailers Trailers to send with the request.
    * @param responseHandler the callback for receiving stream events.
-   * @return CancellableStream, a cancelable request.
+   * @return CancelableStream, a cancelable request.
    */
-  fun send(request: Request, body: ByteBuffer?, trailers: Map<String, List<String>>,
-                responseHandler: ResponseHandler): CancellableStream
+  fun send(request: Request, data: ByteBuffer?, trailers: Map<String, List<String>>,
+                responseHandler: ResponseHandler): CancelableStream
 
   /**
    * Convenience function for sending a unary request.
    *
    * @param request The request to send.
    * @param body Serialized data to send as the body of the request.
    * @param responseHandler the callback for receiving stream events.
-   * @return CancellableStream, a cancelable request.
+   * @return CancelableStream, a cancelable request.
    */
   fun send(request: Request, body: ByteBuffer?,
```

Comment:
Is the only reason for this function to allow consumers to avoid specifying trailers?

Reply:
This is equivalent to this, right: https://github.com/lyft/envoy-mobile/blob/537be8e3a58ad4d17baeef18e727b2171e84c30d/library/swift/src/Envoy.swift#L50

Reply:
Sorry, to clarify: You have 3 functions instead of the two here: https://github.com/lyft/envoy-mobile/blob/537be8e3a58ad4d17baeef18e727b2171e84c30d/library/swift/src/Client.swift

Reply:
We can do an extension function but I don't think the Java consumer would get something that is expected

The other option is to have a default value but we'll have to change the signature to have the trailers be the last parameter (which is kind of lame because I like that the `ResponseHandler` is last)